packages/api/src/command/medical/tcm-encounter/get-tcm-encounters.ts

```diff
@@ -1,88 +1,109 @@
-};
+export interface TcmEncounterQueryData extends TcmEncounterModel {
+  dataValues: TcmEncounterModel["dataValues"] & {
+    patient_data: PatientModel["dataValues"]["data"];
+  };
+}
 
-type TcmEncounterForDisplay = TcmEncounter & {
-  patientName: string;
+export type TcmEncounterResult = TcmEncounterModel["dataValues"] & {
+  patientData: PatientModel["dataValues"]["data"];
 };
 
 export async function getTcmEncounters({
   cxId,
   after,
   pagination,
-}: GetTcmEncountersCmd): Promise<TcmEncounterForDisplay[]> {
-  const where: WhereOptions<TcmEncounterModel> = {
-    cxId,
-    admitTime: {
-      [Op.gt]: after ? buildDayjs(after).toDate() : DEFAULT_FILTER_DATE,
-    },
-  };
+}: {
+  cxId: string;
+  after?: string;
+  pagination: Pagination;
+}): Promise<TcmEncounterResult[]> {
+  const tcmEncounterTable = TcmEncounterModel.tableName;
+  const patientTable = PatientModel.tableName;
 
-  const rows = await TcmEncounterModel.findAll({
-    where,
-    include: [
-      {
-        model: PatientModel,
-        as: "PatientModel",
-        attributes: ["id", "cxId", "data"],
-      },
-    ],
-    limit: pagination.count + 1, // Get one extra to determine if there's a next page
-    order: [["admitTime", "DESC"]],
-  });
+  const sequelize = TcmEncounterModel.sequelize;
+  if (!sequelize) throw new Error("Sequelize not found");
 
-  const items = rows.map((row: TcmEncounterModel) => {
-    const patient = (row.get("PatientModel") as PatientModel).dataValues.data;
+  const { toItem, fromItem } = pagination;
 
-    /** Hack to get around Sequelize type inference not seeing associations */
-    const encounterData = { ...row.dataValues, PatientModel: undefined };
-    delete encounterData.PatientModel;
+  /**
+   * ⚠️ Always change this query and the count query together.
+   */
+  const queryString = `
+      SELECT tcm_encounter.*, patient.data as patient_data
+      FROM ${tcmEncounterTable} tcm_encounter
+      INNER JOIN ${patientTable} patient 
+      ON tcm_encounter.patient_id = patient.id
+      WHERE tcm_encounter.cx_id = :cxId
+      AND tcm_encounter.admit_time > :afterDate
+      ${toItem ? ` AND tcm_encounter.id >= :toItem` : ""}
+      ${fromItem ? ` AND tcm_encounter.id <= :fromItem` : ""}
+      ORDER BY tcm_encounter.id ${toItem ? "ASC" : "DESC"}
+      LIMIT :count
+    `;
 
-    return {
-      ...encounterData,
-      patientName: patient.firstName + " " + patient.lastName,
-      patientDateOfBirth: patient.dob,
-      patientPhoneNumbers: patient.contact?.map(contact => contact.phone) ?? [],
-      patientStates: patient.address?.map(address => address.state) ?? [],
-    };
-  });
+  /**
+   * Use a type assertion to cast the additional data from the join into the return type.
+   */
+  const rawEncounters = (await sequelize.query(queryString, {
+    model: TcmEncounterModel,
+    mapToModel: true,
+    replacements: {
+      cxId,
+      ...{ afterDate: after ? buildDayjs(after).toDate() : DEFAULT_FILTER_DATE },
```

Comment:
Can we always use `.toISOString` instead of `.toDate` when working with dates as DB query params? It sets a pattern that works regardless of how we map the date column (Date or string).

Reply:
yep, this is a TIL, thanks!

Reply:
Cool. Let's also update how we use/set `DEFAULT_FILTER_DATE` to match it? I'd actually create it as dayjs so we can perform operations on it, and when we want to use it we just `.toISOString()` on it